### PR TITLE
Add audit log

### PR DESCRIPTION
**Pull request by liyaqin1**
related PR: https://github.com/pinternal/teletraanconfigs/pull/25

### CODE REVIEW

**Comment by ntascii** [comment]
when username is rodimus?

**Comment by liyaqin1** [comment]
> when username is rodimus?

it means the call is from rodimus api to teletraan api.

**Comment by liyaqin1** [comment]
> when username is rodimus?

see this:
```
teletraandb-1-144:3306 deploy kxadmin1804(admin)> select * from tokens_and_roles where script_name="rodimus";
+-------------+-------------+---------------+------------------------+----------+---------------+
| script_name | resource_id | resource_type | token                  | role     | expire_date   |
+-------------+-------------+---------------+------------------------+----------+---------------+
| rodimus     | *           | SYSTEM        | IPnZCQHNTeCpGHZlZG5BBw | OPERATOR | 1747513917000 |
+-------------+-------------+---------------+------------------------+----------+---------------+
1 row in set (0.00 sec)

teletraandb-1-144:3306 deploy kxadmin1804(admin)> 
```

**Comment by robbintt** [comment]
Hi Yaqin, looks like you put the secret in the ticket? If so can you please
rotate it?

Thanks,
Trent

**Comment by tylerwowen** [comment]
Can you please also include the config updates so that audit log are logged in different files?

File: deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/security/TokenAuthFilter.java [hunk 1]
```diff
@@ -52,9 +52,9 @@ public TokenAuthFilter(String userDataUrl, String groupDataUrl, String userNameK
     public void filter(ContainerRequestContext context) throws IOException {
         if(!context.getMethod().equals("OPTIONS")) {
             SecurityContext securityContext;
-
             try {
                 securityContext = authenticate(context);
+                LOG.info("request method: " + context.getMethod() + "; request uri: " + context.getUriInfo().getRequestUri() + "; user name: " + securityContext.getUserPrincipal().getName());
```
Comment (tylerwowen) [inline deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/security/TokenAuthFilter.java:57]: Could you format the string in json format so it's easier to query when needed? 
i.e.
```
LOG.info("\"method\":{}, \"user\": {}", method, user);
```

**Comment by liyaqin1** [comment]
tested on control, looks good:

```
INFO  [2023-05-09 22:21:38,491] com.pinterest.teletraan.security.TokenAuthFilter: Authentication failed. Reason: java.io.IOException: Server returned HTTP response code: 401 for URL: https://auth.pinadmin.com/api/me/?access_token=eyJhbGciOiJSUzI1NiIsImtpZCI6Ijc2NDc3OTcwODAzODM0NDk4MSIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJGQTR2V2tiZEs4R0hxbFh6Und4ZnU2MlhKcTh0d0lOVjhMbGJGcnpTIiwiZXhwIjoxNjgzNjY3ODg3LCJncm91cHMiOlsiZW5naW5lZXJpbmciLCJpbmZyYS1zZWMiLCJkZWxpdmVyeS10ZWFtIiwianVweXRlci1hY2Nlc3MiLCJidXNpbmVzcy1zZW5zaXRpdmUtZHluYW1pYy1vbGQiLCJidXNpbmVzcy1zZW5zaXRpdmUtc3RhdGljIiwiZW5nLXByb2QiLCJqdXB5dGVyLWFjY2Vzcy10ZXN0IiwianVweXRlci1hY2Nlc3MtZHludGVtcCIsInRyYW5zcGFyZW5jeS1lbmctYWxsIiwiZ3BuLWJ1c2luZXNzLXNlbnNpdGl2ZS1keW5hbWljIiwiYnVzaW5lc3Mtc2Vuc2l0aXZlLWR5bmFtaWMiLCJncG4tYnVzaW5lc3Mtc2Vuc2l0aXZlIiwiY2RwIl0sImlhdCI6MTY4MzA2Mjc4NywiaXNzIjoiaHR0cHM6Ly9hdXRoLnBpbmFkbWluLmNvbSIsImp0aSI6IjdjMDk2MGE0LWIyODQtNDExNy1iZDVhLTYzN2QzY2FiMDg3MyIsIm5iZiI6MTY4MzA2Mjc4Nywic2NwIjpbInVzZXIiXSwic3ViIjoieWFxaW5saSJ9.jm4tPDPpL7fWdoINkakilrMk4QOFugDvhfdGSfbQLRmrfO4LverDrhCOGRXV7UoxKpXYhVmOL8I9Vz58mNCMAutfKh8gUGMzCzWcjymjy85c1lEuJcCs3xfNEwMbyeeL5uKjoAHeN1yeF7DOjcH-oB6Voe8TUrvFwXhTv1APGhfSuMh-oPCzJCLrunNZw7d03e0iBJrU-inEm0I3tdAckiG7jNHkRA2ir-ZWF3am43CKNA5p4cie0nUxel8uGihNbHE6CGRg2bt7EuaROtye2GWF93WoJ-dYCgtSzCv76yPgiIat2aO90yzZ58bffabLnZEXhlTsIRXTDTc9wo1g6g
INFO  [2023-05-09 22:21:48,063] com.pinterest.teletraan.security.UserDataHelper: Retrieved username yaqinli from token.
INFO  [2023-05-09 22:21:48,103] com.pinterest.teletraan.security.UserDataHelper: Retrieved groups [engineering, infra-sec, delivery-team, jupyter-access, business-sensitive-dynamic-old, business-sensitive-static, eng-prod, jupyter-access-test, jupyter-access-dyntemp, transparency-eng-all, gpn-business-sensitive-dynamic, business-sensitive-dynamic, gpn-business-sensitive, cdp] from token.
INFO  [2023-05-09 22:21:48,104] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap; user name: yaqinli}
INFO  [2023-05-09 22:21:48,104] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap; user name: yaqinli}
INFO  [2023-05-09 22:21:48,165] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap/promotes; user name: yaqinli}
INFO  [2023-05-09 22:21:48,165] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: PUT; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap/deploys/current/progress; user name: yaqinli}
INFO  [2023-05-09 22:21:48,198] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/deploys/er5pARFhRpCLOyaR8Ft5wQ; user name: yaqinli}
INFO  [2023-05-09 22:21:48,220] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/builds/GQuTCBsWSeG9aqJGTXSq4Q_be7fa52; user name: yaqinli}
INFO  [2023-05-09 22:21:48,239] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/builds/tags?pageIndex=1&name=teletraan-service&pageSize=5; user name: yaqinli}
INFO  [2023-05-09 22:21:48,253] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/deploys/er5pARFhRpCLOyaR8Ft5wQ; user name: yaqinli}
INFO  [2023-05-09 22:21:48,268] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/builds/GQuTCBsWSeG9aqJGTXSq4Q_be7fa52/tags; user name: yaqinli}
INFO  [2023-05-09 22:22:06,482] com.pinterest.teletraan.security.UserDataHelper: Retrieved username yaqinli from token.
INFO  [2023-05-09 22:22:06,540] com.pinterest.teletraan.security.UserDataHelper: Retrieved groups [engineering, infra-sec, delivery-team, jupyter-access, business-sensitive-dynamic-old, business-sensitive-static, eng-prod, jupyter-access-test, jupyter-access-dyntemp, transparency-eng-all, gpn-business-sensitive-dynamic, business-sensitive-dynamic, gpn-business-sensitive, cdp] from token.
INFO  [2023-05-09 22:22:06,540] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: https://teletraan-control.pinadmin.com/swagger.json; user name: yaqinli}
INFO  [2023-05-09 22:22:48,040] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap; user name: yaqinli}
INFO  [2023-05-09 22:22:48,040] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap; user name: yaqinli}
INFO  [2023-05-09 22:22:48,055] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap/promotes; user name: yaqinli}
INFO  [2023-05-09 22:22:48,059] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: PUT; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap/deploys/current/progress; user name: yaqinli}
INFO  [2023-05-09 22:22:48,070] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/deploys/er5pARFhRpCLOyaR8Ft5wQ; user name: yaqinli}
INFO  [2023-05-09 22:22:48,084] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/builds/GQuTCBsWSeG9aqJGTXSq4Q_be7fa52; user name: yaqinli}
INFO  [2023-05-09 22:22:48,099] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/builds/tags?pageIndex=1&name=teletraan-service&pageSize=5; user name: yaqinli}
INFO  [2023-05-09 22:22:48,116] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/deploys/er5pARFhRpCLOyaR8Ft5wQ; user name: yaqinli}
INFO  [2023-05-09 22:22:48,130] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/builds/GQuTCBsWSeG9aqJGTXSq4Q_be7fa52/tags; user name: yaqinli}
INFO  [2023-05-09 22:23:44,047] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: https://teletraan-control.pinadmin.com/v1/envs?envName=teletraan; user name: yaqinli}
INFO  [2023-05-09 22:23:48,028] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap; user name: yaqinli}
INFO  [2023-05-09 22:23:48,033] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap; user name: yaqinli}
INFO  [2023-05-09 22:23:48,045] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap/promotes; user name: yaqinli}
INFO  [2023-05-09 22:23:48,048] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: PUT; request uri: http://teletraan-control.pinadmin.com/v1/envs/teletraan/bootstrap/deploys/current/progress; user name: yaqinli}
INFO  [2023-05-09 22:23:48,060] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/deploys/er5pARFhRpCLOyaR8Ft5wQ; user name: yaqinli}
INFO  [2023-05-09 22:23:48,074] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/builds/GQuTCBsWSeG9aqJGTXSq4Q_be7fa52; user name: yaqinli}
INFO  [2023-05-09 22:23:48,089] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/builds/tags?pageIndex=1&name=teletraan-service&pageSize=5; user name: yaqinli}
INFO  [2023-05-09 22:23:48,105] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/deploys/er5pARFhRpCLOyaR8Ft5wQ; user name: yaqinli}
INFO  [2023-05-09 22:23:48,116] com.pinterest.teletraan.security.TokenAuthFilter: Audit log. Details: {request method: GET; request uri: http://teletraan-control.pinadmin.com/v1/builds/GQuTCBsWSeG9aqJGTXSq4Q_be7fa52/tags; user name: yaqinli}

```

**Review by tylerwowen — APPROVED** [review]
nit: we shouldn't have space in key.

**Comment by liyaqin1** [comment]
> nit: we shouldn't have space in key.

Thanks. but we donn't have space in key ^^

**Comment by liyaqin1** [comment]
```
INFO  [2023-05-10 19:28:03,191] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/envs/helloworlddummyservice-server-dev1/aagnese-test/capacity?capacityType=GROUP", "user name": "rodimus"}
INFO  [2023-05-10 19:28:03,225] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/envs/2ceglCGUS9-FIHBNfaYqbA", "user name": "rodimus"}
INFO  [2023-05-10 19:28:03,239] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/envs/helloworlddummyservice-server-dev1/test-tyler-2/capacity?capacityType=GROUP", "user name": "rodimus"}
INFO  [2023-05-10 19:28:15,142] com.pinterest.teletraan.security.TokenAuthFilter: Authentication failed. Reason: Can not find HTTP header: Authorization!
INFO  [2023-05-10 19:28:42,662] com.pinterest.teletraan.security.UserDataHelper: Retrieved username yaqinli from token.
INFO  [2023-05-10 19:28:42,699] com.pinterest.teletraan.security.UserDataHelper: Retrieved groups [engineering, infra-sec, delivery-team, jupyter-access, business-sensitive-dynamic-old, business-sensitive-static, eng-prod, jupyter-access-test, jupyter-access-dyntemp, transparency-eng-all, gpn-business-sensitive-dynamic, business-sensitive-dynamic, gpn-business-sensitive, cdp] from token.
INFO  [2023-05-10 19:28:42,699] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "https://teletraan-dev1.pinadmin.com/v1/envs?envName=helloworlddummyservice-server", "user name": "yaqinli"}
INFO  [2023-05-10 19:28:50,844] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/deploys?pageIndex=1&pageSize=10&deployState=RUNNING&deployState=FAILING", "user name": "yaqinli"}
INFO  [2023-05-10 19:28:50,845] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/envs/sidecars", "user name": "yaqinli"}
INFO  [2023-05-10 19:28:50,861] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/envs/akT9t2RETkitoVYEVPtELg", "user name": "yaqinli"}
INFO  [2023-05-10 19:28:50,864] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/deploys?envId=bqmJ5yB9QpCOPnx_3AI3UQ&envId=QQX1n7zkTHqsdI9I3b7CVQ&envId=SHdhDpsdTheNKeaIXmi2hw&envId=WUJjjYDFRyC7t36567q5pQ&envId=XPQD3SCpRBSDnE_wHgzJzw&deployState=RUNNING&deployState=FAILING", "user name": "yaqinli"}
INFO  [2023-05-10 19:28:50,875] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/builds/nZgqyW_gSSS0kx8fqUg7yw_41f64c9", "user name": "yaqinli"}
INFO  [2023-05-10 19:28:50,891] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/envs/f3-QBcdtQ8CZVmaG0p7lSw", "user name": "yaqinli"}
INFO  [2023-05-10 19:28:50,907] com.pinterest.teletraan.security.TokenAuthFilter: {"request method": "GET", "request uri": "http://teletraan-dev1.pinadmin.com/v1/builds/eLcP1QNXS063VpwZ8j2sSA_f70110a", "user name": "yaqinli"}
```

**Pull request closed without merging**